### ISSUE TITLE
New Repo: `func-python`

### ISSUE THREAD

**Issue by matzew** [knative/community#1558]
Use this issue type to request a new repo in `knative-extensions` (or
`knative`, which may require additional discussion).

## Repo information

Org: `knative-extensions`

Repo: `func-python`

Purpose (Description):

```
The repository contains the function invocation and runtime framework for Knative Python functions.
```

Sponsoring WG:
Knative Functions/Clients

## Actions to fulfill

This area is used to track the [repo creation process](https://github.com/knative/community/blob/main/mechanics/CREATING-AN-EXTENSIONS-REPO.md).
The _requestor_ and _sponsoring WG lead_ should perform the steps listed below and cross out the checkmarks when done.
The TOC is involved only in the **TOC Gate** steps.

- [x] Add this issue to the [TOC project board](https://github.com/orgs/knative/projects/43) for review. You are responsible for moving your entry on the board to "Needs Discussion" or "In Progress" as you move forward in this checklist.

_You may not be able to use the Projects quick menu on this page. In that case, go to the project board and use the **Add cards** interface._

- [x] Send a PR adding entries for this repo in [`/peribolos/knative-extensions.yaml`](https://github.com/knative/community/blob/main/peribolos/knative-extensions.yaml). Please mind the alphabetical order when adding to a list.
  - [x] Add the repository and a description.
  - [x] Grant `Knative Admin` the `admin` privilege.
  - [x] Grant the sponsoring WG the `write` privilege.

**TOC Gate**: _Once the TOC has approved the above, it will merge and Peribolos will create an empty repository._

- [x] Have a lead from the sponsoring WG bootstrap the Git repository by using an 
  appropriate "template" repository ([basic](https://github.com/knative-extensions/wg-repository),
  [sample-controller](https://github.com/knative-extensions/sample-controller),
  [sample-source](https://github.com/knative-extensions/sample-source)).

  Follow the instructions from [Creating a repository from a template](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-repository-from-a-template)

- [x] Send a PR adding the repo to [knobots](https://github.com/knative-extensions/knobots). Then run the community files update action to propagate new OWNER_ALIASES.

- [x] Add your GitHub Team to the `OWNERS` file for your repo.

- [x] [Set up](https://github.com/knative/test-infra/blob/main/guides/prow_knative_setup.md#setting-up-prow-for-a-new-repo-reviewers-assignment-and-auto-merge) prow for a new repo

- [ ] Bootstrap your CI jobs using [hack](https://github.com/knative/hack) project (look at other extensions repos for reference)

- [x] Create a sample PR to verify Prow (e.g., edit the boilerplate README)

- [x] Verify that within 24 hours the appropriate branch protections have been applied
   requiring `tide` to pass before PRs are merged.

- [ ] (optional) Send a PR adding the repo to [CLOMonitor](https://clomonitor.io/projects/cncf/knative) ([more information](/REPOSITORY-GUIDELINES.md#clomonitor-and-clotributor)).

**Comment by dsimansk** [comment]
+1 as TOC

**Comment by dprotaso** [comment]
+1 as TOC

**Comment by psschwei** [comment]
+1

**Comment by lkingland** [comment]
+1
Thanks for getting the ball rolling Matthias.  Description looks good to me.

**Comment by dprotaso** [comment]
@matzew just following up here if you're blocked on anything

**Comment by lkingland** [comment]
I think we're getting close:

> [ ] Have a lead from the sponsoring WG bootstrap ....
The repo appears to have been bootstrapped so this is perhaps done?

> [ ] Send a PR adding the repo to knobots.
I believe [this PR](https://github.com/knative-extensions/knobots/pull/383) is the one?  So it's complete once the hold is removed.

> [ ] Set up Prow
I [see it's enabled](https://github.com/knative-extensions/func-python/pull/1#issuecomment-2360025358), but according to [the instructions](https://github.com/knative/test-infra/blob/main/guides/prow_knative_setup.md#setting-up-prow-for-a-new-repo-reviewers-assignment-and-auto-merge), we have a few more steps to complete.  In particular is it a problem I see no configurations for the repo [in the generated configs](https://github.com/knative/test-infra/tree/main/prow/jobs/generated/knative-sandbox)? 

> [ ] Bootstrap your CI jobs using [hack](https://github.com/knative/hack) project (look at other extensions repos for reference)
I think this still needs to be done.   I can add these in a separate pr using func-go as a template.

> [ ] Create a sample PR to verify Prow (e.g., edit the boilerplate README)
This is accomplished by [PR#1](https://github.com/knative-extensions/func-python/pull/1)

> [ ] Verify that within 24 hours the appropriate branch protections have been applied
requiring tide to pass before PRs are merged.
This also appears to be done because PR#1 is not able to be merged without all required statuses being passed.

**Comment by lkingland** [comment]
Hey @matzew the repository now appears to be fully working other than adding the /hack scripts.  Do you know if they are necessary?  If not, we can close this issue as completed 👍🏻

**Comment by matzew** [comment]
@lkingland I was just checking the same :smile:
